File: Novels/packages/5-mobile-ios/StoryReader/Model/Story.swift
import Foundation

// Story model - Converted from 7-shared StoryDto
struct Story: Codable, Identifiable {
    let id: String
    let title: String
    let author: String
    let description: String?
    let coverImage: String?
    let genreId: Int?
    let genre: String?
    let status: String?
    let totalChapters: Int?
    let createdAt: Date?
    let updatedAt: Date?
    
    enum CodingKeys: String, CodingKey {
        case id, title, author, description
        case coverImage = "cover_image"
        case genreId = "genre_id"
        case genre, status
        case totalChapters = "total_chapters"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Story.kt
package com.storysphere.storyreader.model

data class Story(
    val id: String,
    val title: String,
    val author: String,
    val description: String? = null,
    val coverImage: String? = null,
    val genreId: Int? = null,
    val genre: String? = null,
    val status: StoryStatus = StoryStatus.ONGOING,
    val totalChapters: Int = 0,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

enum class StoryStatus {
    ONGOING,
    COMPLETED,
    HIATUS
}

